Source/Parallelization/WarpXComm.cpp

```diff
@@ -341,141 +396,158 @@ WarpX::UpdateAuxilaryDataSameType ()
-                Array4<Real const> const& bx_fp = Bfield_fp[lev][0]->const_array(mfi);
-                Array4<Real const> const& by_fp = Bfield_fp[lev][1]->const_array(mfi);
-                Array4<Real const> const& bz_fp = Bfield_fp[lev][2]->const_array(mfi);
-                Array4<Real const> const& bx_c = dBx.const_array(mfi);
-                Array4<Real const> const& by_c = dBy.const_array(mfi);
-                Array4<Real const> const& bz_c = dBz.const_array(mfi);
-
-                amrex::ParallelFor(Box(bx_aux), Box(by_aux), Box(bz_aux),
-                [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
-                {
-                    warpx_interp(j, k, l, bx_aux, bx_fp, bx_c, Bx_stag, refinement_ratio);
-                },
-                [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
-                {
-                    warpx_interp(j, k, l, by_aux, by_fp, by_c, By_stag, refinement_ratio);
-                },
-                [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
+                for (MFIter mfi(*Bfield_aux[lev][0]); mfi.isValid(); ++mfi)
                 {
-                    warpx_interp(j, k, l, bz_aux, bz_fp, bz_c, Bz_stag, refinement_ratio);
-                });
+                    Array4<Real> const& bx_aux = Bfield_aux[lev][0]->array(mfi);
+                    Array4<Real> const& by_aux = Bfield_aux[lev][1]->array(mfi);
+                    Array4<Real> const& bz_aux = Bfield_aux[lev][2]->array(mfi);
+                    Array4<Real const> const& bx_fp = Bfield_fp[lev][0]->const_array(mfi);
+                    Array4<Real const> const& by_fp = Bfield_fp[lev][1]->const_array(mfi);
+                    Array4<Real const> const& bz_fp = Bfield_fp[lev][2]->const_array(mfi);
+                    Array4<Real const> const& bx_c = dBx.const_array(mfi);
+                    Array4<Real const> const& by_c = dBy.const_array(mfi);
+                    Array4<Real const> const& bz_c = dBz.const_array(mfi);
+
+                    amrex::ParallelFor(Box(bx_aux), Box(by_aux), Box(bz_aux),
+                    [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
+                    {
+                        warpx_interp(j, k, l, bx_aux, bx_fp, bx_c, Bx_stag, refinement_ratio);
+                    },
+                    [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
+                    {
+                        warpx_interp(j, k, l, by_aux, by_fp, by_c, By_stag, refinement_ratio);
+                    },
+                    [=] AMREX_GPU_DEVICE (int j, int k, int l) noexcept
+                    {
+                        warpx_interp(j, k, l, bz_aux, bz_fp, bz_c, Bz_stag, refinement_ratio);
+                    });
+                }
+            }
+            else // electrostatic
+            {
+                MultiFab::Copy(*Bfield_aux[lev][0], *Bfield_fp[lev][0], 0, 0, Bfield_aux[lev][0]->nComp(), Bfield_aux[lev][0]->nGrowVect());
```

Comment:
We may need to check whether this needs to use `Bfield_aux[lev][0]->nGrowVect()` or `ng_gather`...

Reply:
@WeiqunZhang 
Speaking of ghost cells, do you think it is correct to use this number of ghost cells here? We are calling `Copy` because we essentially need to copy the `_fp` arrays into the `_aux` ones. What happens, for instance, if the source MultiFab has less ghost cells than the ones that we pass here, taken from the destination MultiFab?

Reply:
That would be an error, if `_aux` has more ghost cells than `_fp`.

Reply:
Thanks!

In such a case, what would be your advice if one is not sure about the exact number of ghost cells required by the specific algorithm implemented here? 

Would it be more robust to call `Copy` by passing a number of ghost cells that is the minimum between the destination and source ghost cells?

Or would it be more robust to leave it as is, relying on AMReX to flag (with either an assertion error or a segfault - depending what is implemented on the AMReX side) if the destination ghost cells are more than the source ghost cells, thus pointing out an underlying bug on the WarpX side?